gpu: amd: add support for rocm6.1 (#2662)
oneapi-src/oneDNN: src/gpu/amd/README.md

## General information

Support for AMD backend is implemented via SYCL HIP backend. The feature is

> disabled by default and is currently experimental. Users must enable it at build

Reviewer: ```suggestion
disabled by default and is currently experimental. You must enable it at the build
```

Author: Done, thanks.

time with a CMake option `DNNL_GPU_VENDOR=AMD`. The AMD GPUs can be used via
oneDNN engine abstraction. The engine should be created using

> `dnnl::engine::kind::gpu` engine kind or the user can provide a `sycl::device`

Reviewer: ```suggestion
`dnnl::engine::kind::gpu` engine kind or you can provide `sycl::device`
```

> objects that corresponds to AMD GPUs.

Reviewer: ```suggestion
objects that correspond to AMD GPUs.
```

Author: Done, thanks.
